README.md

```diff
@@ -1,22 +1,48 @@
 2023/03/01
 Hsiang-He Lee (lee1061@llnl.gov)
+2023/05/24 updated
 
-E3SM chemistry diagnostic package version1.0 (ChemDyg v1.0) is designed for E3SM chemistry development. There are total 11 types of plot and 3 types of table for model-to-model and model-to-observation comparison.
+E3SM chemistry diagnostic package version 0.1.0 (ChemDyg v0.1.0) is designed for E3SM chemistry development. There are total 11 types of plot and 3 types of table for model-to-model and model-to-observation comparison.
 
 Index page and example figures:
 https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.lee1061/20220914.PAN.MZThet.v2.LR.bi-grid.amip.chemUCI_Linozv3/e3sm_chem_diags/plots/
 
 ChemDyg is executed by zppy, which is a post-processing toolchain for E3SM written in Python (https://github.com/E3SM-Project/zppy).
 
-*How to run ChemDyg via zppy script on chrysalis?
-1. Setup environment 
->source /gpfs/fs1/soft/chrysalis/manual/anaconda3/2020.11/etc/profile.d/conda.sh
->conda activate zppy_dev_chem
+# How to run ChemDyg via zppy script on chrysalis?
 
-2. Download zppy and ChemDyg on your working folder 
+*Clone repository
 
-3. Modify .cfg run script and then
->zppy -c ChemDyg_example_script.cfg
+```git clone git@github.com:E3SM-Project/zppy.git```
+
+```cd zppy```
+
+*Checkout branch
+```git checkout golaz/plugins```
+
+*Create environment
+
+```module load anaconda3/2020.11```
+
+```conda env create --name zppy_dev_chem -f conda/dev.yml```
+
+*Activate
+
+```source /gpfs/fs1/soft/chrysalis/manual/anaconda3/2020.11/etc/profile.d/conda.sh```
+
+```conda activate zppy_dev_chem```
+
+*Install chemdyg package
+
+```install -y -c e3sm/label/chemdyg_dev chemdyg```
+
+*Install zppy in environment
+
+```python -m pip install .```
+
+*Once this is done, you call is as
+
+```zppy -c ChemDyg_example_script.cfg```
```

Comment:
@xylar , this part of ChemDyg instructions will need to be updated when zppy's next release includes the plugin feature. We feel it makes more sense to this following. Is there a standard way? Any suggests are welcome. Thanks.

- Leave a link to the instructions in this README.md file.
- Create another file in this ChemDyg github repo with step-by-step instructions, to which the link above points.
- Exclude the link file from the python package. Do we do this in MANIFEST.in?

Reply:
I see how this is tricky.

Ultimately, how you want to document the package, including installation and usage instructions, is up to you.

It would be standard to include these instructions directly in the README.md, so that's what I would recommend.  It is okay if the instructions aren't correct at the time of the package's release (though ideally they would be).  Users will not have easy access to this file in the python package so it should not be assumed to be the way that they will know how to run the code.

It is also okay if you update the README.md (or whatever method you have for providing users with installation and usage instructions) after the release, since most users would presumably come to the GitHub repo and look at the latest README.md for instructions (unless you make a formal documentation page somewhere else).

Reply:
I think it would be better to edit these instructions to be as close as possible to the correct ones *after* zppy and chemdyg are released, rather than to provide the current testing instructions.

Reply:
Agreed. It is good to know that users won't have easy access to the README.md file in the release version and will mainly rely on this GitHub repo for instructions. That is exactly what we want - The instructions are somewhat independent of the release. Perfect!

@hsiangheleellnl , can you revise this README.md as suggested by @xylar below and create a draft release after that? We will then iterate on the release documentation.